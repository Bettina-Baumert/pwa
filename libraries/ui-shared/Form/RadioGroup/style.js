import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const container = css({
  paddingTop: themeConfig.variables.gap.big * 1.5,
  display: 'flex',
  flexDirection: 'column',
}).toString();

const label = css({
  fontSize: '0.8rem',
  paddingBottom: themeConfig.variables.gap.small,
  color: themeConfig.colors.shade12,
}).toString();

export default {
  container,
  label,
};
