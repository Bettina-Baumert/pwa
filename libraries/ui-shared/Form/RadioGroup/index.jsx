import React, { Component, Children, cloneElement } from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import FormElement from '@shopgate/pwa-ui-shared/FormElement';
import style from './style';

/**
 * RadioGroup component.
 */
class RadioGroup extends Component {
  static propTypes = {
    children: PropTypes.node,
    default: PropTypes.string,
    errorText: PropTypes.string,
    label: PropTypes.string,
    onChange: PropTypes.func,
  }

  static defaultProps = {
    onChange: () => {},
    children: null,
    errorText: '',
    label: '',
    default: null,
  }

  /**
   * Initializes the component.
   * @param {Object} props The components props.
   */
  constructor(props) {
    super(props);
    this.state = { active: props.default };
  }

  /**
   * Renders the component.
   * @returns {JSX}
   */
  render() {
    const {
      children, onChange, label, errorText,
    } = this.props;

    return (
      <FormElement
        label={label}
        errorText={errorText}
        htmlFor="none"
        hasUnderline={false}
        hasValue
      >
        <div className={style.container}>
          {Children.map(children, child => cloneElement(child, {
            active: this.state.active === child.props.name,
            onChange: () => {
              this.setState({ active: child.props.name });
              onChange(child.props.name);
            },
          }))}
        </div>
      </FormElement>
    );
  }
}

export default RadioGroup;
