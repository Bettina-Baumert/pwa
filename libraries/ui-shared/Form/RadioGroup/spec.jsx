import React from 'react';
import { mount } from 'enzyme';
import RadioItem from './components/Item';
import RadioGroup from '.';

describe('<RadioGroup />', () => {
  it('should render empty group', () => {
    const wrapper = mount(<RadioGroup />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render group with item', () => {
    const wrapper = mount((
      <RadioGroup>
        <RadioItem name="foo" label="foo" />
      </RadioGroup>
    ));

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(RadioItem).length).toEqual(1);
    expect(typeof wrapper.find(RadioItem).prop('onChange')).toEqual('function');
  });

  it('should make default as active', () => {
    const wrapper = mount((
      <RadioGroup default="foo">
        <RadioItem name="foo" label="foo" />
      </RadioGroup>
    ));

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(RadioItem).length).toEqual(1);
    expect(wrapper.find(RadioItem).prop('active')).toEqual(true);
  });

  it('should have on active at a time', () => {
    const wrapper = mount((
      <RadioGroup>
        <RadioItem name="foo" label="foo" />
        <RadioItem name="bar" label="bar" />
      </RadioGroup>
    ));

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(RadioItem).length).toEqual(2);

    // First element active
    expect(wrapper.find(RadioItem).at(0).prop('onChange')());
    wrapper.update();
    expect(wrapper.find(RadioItem).at(0).prop('active')).toEqual(true);
    expect(wrapper.find(RadioItem).at(1).prop('active')).toEqual(false);

    // Second element active
    expect(wrapper.find(RadioItem).at(1).prop('onChange')());
    wrapper.update();
    expect(wrapper.find(RadioItem).at(0).prop('active')).toEqual(false);
    expect(wrapper.find(RadioItem).at(1).prop('active')).toEqual(true);
  });

  it('should call onChange callback', () => {
    const onChange = jest.fn();
    const wrapper = mount((
      <RadioGroup onChange={onChange}>
        <RadioItem name="foo" label="foo" />
      </RadioGroup>
    ));

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(RadioItem).length).toEqual(1);
    wrapper.find(RadioItem).prop('onChange')();

    expect(onChange).toHaveBeenCalledWith('foo');
  });
});
