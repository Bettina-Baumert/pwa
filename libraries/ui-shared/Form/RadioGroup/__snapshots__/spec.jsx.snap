// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> should call onChange callback 1`] = `
<RadioGroup
  default={null}
  errorText=""
  label=""
  onChange={[MockFunction]}
>
  <FormElement
    className=""
    errorText=""
    hasPlaceholder={true}
    hasUnderline={false}
    hasValue={true}
    htmlFor="none"
    isFocused={false}
    label=""
    placeholder=""
  >
    <div
      className="css-1vjh8yv "
    >
      <Placeholder
        placeholder=""
        visible={false}
      >
        <div
          className="css-oc0mp css-1jyyyen"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </div>
      </Placeholder>
      <Label
        hasErrorMessage={false}
        htmlFor="none"
        isFloating={true}
        isFocused={false}
        label=""
      >
        <label
          className="css-1nn9zl4 css-16msjt6 css-1fekhao"
          htmlFor="none"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </label>
      </Label>
      <div
        className="css-zbxrv7"
      >
        <RadioItem
          active={false}
          key=".0"
          label="foo"
          name="foo"
          onChange={[Function]}
        >
          <label
            className="css-1cqgl9p"
            htmlFor="foo0.30210713149961976"
          >
            <RadioUnchecked>
              <Icon
                className=""
                color={null}
                content="<path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"/>"
                size="inherit"
                viewBox="0 0 24 24"
              >
                <svg
                  className="css-yys9hb "
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"/>",
                    }
                  }
                  style={
                    Object {
                      "fill": null,
                      "fontSize": "inherit",
                    }
                  }
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </Icon>
            </RadioUnchecked>
            <input
              checked={false}
              className="css-gyrv08"
              id="foo0.30210713149961976"
              name="foo"
              onChange={[Function]}
              type="radio"
            />
            <Translate
              className="css-1wx4qok"
              params={Object {}}
              string="foo"
            >
              <span
                className="css-1wx4qok"
              >
                foo
              </span>
            </Translate>
          </label>
        </RadioItem>
      </div>
      <ErrorText
        className={null}
        errorText=""
      >
        <div
          className="css-hmgfge"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </div>
      </ErrorText>
    </div>
  </FormElement>
</RadioGroup>
`;

exports[`<RadioGroup /> should have on active at a time 1`] = `
<RadioGroup
  default={null}
  errorText=""
  label=""
  onChange={[Function]}
>
  <FormElement
    className=""
    errorText=""
    hasPlaceholder={true}
    hasUnderline={false}
    hasValue={true}
    htmlFor="none"
    isFocused={false}
    label=""
    placeholder=""
  >
    <div
      className="css-1vjh8yv "
    >
      <Placeholder
        placeholder=""
        visible={false}
      >
        <div
          className="css-oc0mp css-1jyyyen"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </div>
      </Placeholder>
      <Label
        hasErrorMessage={false}
        htmlFor="none"
        isFloating={true}
        isFocused={false}
        label=""
      >
        <label
          className="css-1nn9zl4 css-16msjt6 css-1fekhao"
          htmlFor="none"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </label>
      </Label>
      <div
        className="css-zbxrv7"
      >
        <RadioItem
          active={false}
          key=".0"
          label="foo"
          name="foo"
          onChange={[Function]}
        >
          <label
            className="css-1cqgl9p"
            htmlFor="foo0.4545178122079825"
          >
            <RadioUnchecked>
              <Icon
                className=""
                color={null}
                content="<path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"/>"
                size="inherit"
                viewBox="0 0 24 24"
              >
                <svg
                  className="css-yys9hb "
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"/>",
                    }
                  }
                  style={
                    Object {
                      "fill": null,
                      "fontSize": "inherit",
                    }
                  }
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </Icon>
            </RadioUnchecked>
            <input
              checked={false}
              className="css-gyrv08"
              id="foo0.4545178122079825"
              name="foo"
              onChange={[Function]}
              type="radio"
            />
            <Translate
              className="css-1wx4qok"
              params={Object {}}
              string="foo"
            >
              <span
                className="css-1wx4qok"
              >
                foo
              </span>
            </Translate>
          </label>
        </RadioItem>
        <RadioItem
          active={false}
          key=".1"
          label="bar"
          name="bar"
          onChange={[Function]}
        >
          <label
            className="css-1cqgl9p"
            htmlFor="bar0.5481175653818215"
          >
            <RadioUnchecked>
              <Icon
                className=""
                color={null}
                content="<path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"/>"
                size="inherit"
                viewBox="0 0 24 24"
              >
                <svg
                  className="css-yys9hb "
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"/>",
                    }
                  }
                  style={
                    Object {
                      "fill": null,
                      "fontSize": "inherit",
                    }
                  }
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </Icon>
            </RadioUnchecked>
            <input
              checked={false}
              className="css-gyrv08"
              id="bar0.5481175653818215"
              name="bar"
              onChange={[Function]}
              type="radio"
            />
            <Translate
              className="css-1wx4qok"
              params={Object {}}
              string="bar"
            >
              <span
                className="css-1wx4qok"
              >
                bar
              </span>
            </Translate>
          </label>
        </RadioItem>
      </div>
      <ErrorText
        className={null}
        errorText=""
      >
        <div
          className="css-hmgfge"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </div>
      </ErrorText>
    </div>
  </FormElement>
</RadioGroup>
`;

exports[`<RadioGroup /> should make default as active 1`] = `
<RadioGroup
  default="foo"
  errorText=""
  label=""
  onChange={[Function]}
>
  <FormElement
    className=""
    errorText=""
    hasPlaceholder={true}
    hasUnderline={false}
    hasValue={true}
    htmlFor="none"
    isFocused={false}
    label=""
    placeholder=""
  >
    <div
      className="css-1vjh8yv "
    >
      <Placeholder
        placeholder=""
        visible={false}
      >
        <div
          className="css-oc0mp css-1jyyyen"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </div>
      </Placeholder>
      <Label
        hasErrorMessage={false}
        htmlFor="none"
        isFloating={true}
        isFocused={false}
        label=""
      >
        <label
          className="css-1nn9zl4 css-16msjt6 css-1fekhao"
          htmlFor="none"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </label>
      </Label>
      <div
        className="css-zbxrv7"
      >
        <RadioItem
          active={true}
          key=".0"
          label="foo"
          name="foo"
          onChange={[Function]}
        >
          <label
            className="css-1cqgl9p"
            htmlFor="foo0.532946595670658"
          >
            <RadioChecked
              className="css-1ncyhu"
            >
              <Icon
                className="css-1ncyhu"
                color={null}
                content="<path d=\\"M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"/>"
                size="inherit"
                viewBox="0 0 24 24"
              >
                <svg
                  className="css-yys9hb css-1ncyhu"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<path d=\\"M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"/>",
                    }
                  }
                  style={
                    Object {
                      "fill": null,
                      "fontSize": "inherit",
                    }
                  }
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </Icon>
            </RadioChecked>
            <input
              checked={true}
              className="css-gyrv08"
              id="foo0.532946595670658"
              name="foo"
              onChange={[Function]}
              type="radio"
            />
            <Translate
              className="css-1wx4qok"
              params={Object {}}
              string="foo"
            >
              <span
                className="css-1wx4qok"
              >
                foo
              </span>
            </Translate>
          </label>
        </RadioItem>
      </div>
      <ErrorText
        className={null}
        errorText=""
      >
        <div
          className="css-hmgfge"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </div>
      </ErrorText>
    </div>
  </FormElement>
</RadioGroup>
`;

exports[`<RadioGroup /> should render empty group 1`] = `
<RadioGroup
  default={null}
  errorText=""
  label=""
  onChange={[Function]}
>
  <FormElement
    className=""
    errorText=""
    hasPlaceholder={true}
    hasUnderline={false}
    hasValue={true}
    htmlFor="none"
    isFocused={false}
    label=""
    placeholder=""
  >
    <div
      className="css-1vjh8yv "
    >
      <Placeholder
        placeholder=""
        visible={false}
      >
        <div
          className="css-oc0mp css-1jyyyen"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </div>
      </Placeholder>
      <Label
        hasErrorMessage={false}
        htmlFor="none"
        isFloating={true}
        isFocused={false}
        label=""
      >
        <label
          className="css-1nn9zl4 css-16msjt6 css-1fekhao"
          htmlFor="none"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </label>
      </Label>
      <div
        className="css-zbxrv7"
      />
      <ErrorText
        className={null}
        errorText=""
      >
        <div
          className="css-hmgfge"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </div>
      </ErrorText>
    </div>
  </FormElement>
</RadioGroup>
`;

exports[`<RadioGroup /> should render group with item 1`] = `
<RadioGroup
  default={null}
  errorText=""
  label=""
  onChange={[Function]}
>
  <FormElement
    className=""
    errorText=""
    hasPlaceholder={true}
    hasUnderline={false}
    hasValue={true}
    htmlFor="none"
    isFocused={false}
    label=""
    placeholder=""
  >
    <div
      className="css-1vjh8yv "
    >
      <Placeholder
        placeholder=""
        visible={false}
      >
        <div
          className="css-oc0mp css-1jyyyen"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </div>
      </Placeholder>
      <Label
        hasErrorMessage={false}
        htmlFor="none"
        isFloating={true}
        isFocused={false}
        label=""
      >
        <label
          className="css-1nn9zl4 css-16msjt6 css-1fekhao"
          htmlFor="none"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </label>
      </Label>
      <div
        className="css-zbxrv7"
      >
        <RadioItem
          active={false}
          key=".0"
          label="foo"
          name="foo"
          onChange={[Function]}
        >
          <label
            className="css-1cqgl9p"
            htmlFor="foo0.3277093784514742"
          >
            <RadioUnchecked>
              <Icon
                className=""
                color={null}
                content="<path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"/>"
                size="inherit"
                viewBox="0 0 24 24"
              >
                <svg
                  className="css-yys9hb "
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"/>",
                    }
                  }
                  style={
                    Object {
                      "fill": null,
                      "fontSize": "inherit",
                    }
                  }
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </Icon>
            </RadioUnchecked>
            <input
              checked={false}
              className="css-gyrv08"
              id="foo0.3277093784514742"
              name="foo"
              onChange={[Function]}
              type="radio"
            />
            <Translate
              className="css-1wx4qok"
              params={Object {}}
              string="foo"
            >
              <span
                className="css-1wx4qok"
              >
                foo
              </span>
            </Translate>
          </label>
        </RadioItem>
      </div>
      <ErrorText
        className={null}
        errorText=""
      >
        <div
          className="css-hmgfge"
        >
          <Translate
            className=""
            params={Object {}}
            string=""
          >
            <span
              className=""
            />
          </Translate>
        </div>
      </ErrorText>
    </div>
  </FormElement>
</RadioGroup>
`;
